pypeclub/OpenPype
# Bugfix/fix slate timecode and rendering range

## Brief description
Timecode was not correctly added when slate was present, nor was correctly offset when skip handle frames option was selected in the ExtractReview tags. Review duration was off for first time renders with slate but ok without.

## Description
Timecodes in review movs (from Nuke) were not correctly updated to cater for extra slate frame on head.
Timecode should shift -1 frame from starting timecode of extracted representation:
This means that each representation (and its frame length, especially in the case of "Skip handle frames" in outupt videos) should always be:
- rendered (the data mov) with the correct timecode at head and the correct length if slate is present
- checked against new lengths in case of trimming
- have a -1 frame shift in timecode in case of slate being present
- updated with new timecodes per representation before concat/merge

This needed additions in _OpenPype/lib/editorial.py_ with function to deal with timecode and frames which are now available in `openpype.lib` package, plus Logic fixes in the slate extraction, extract review and integrate new.

This should fix problems mentioned in this issue:
[Nuke: Review with Slate doesn't have source timecode #2913 ](https://github.com/pypeclub/OpenPype/issues/2913)
Still needs silent audio in slate as per another PR though

## Testing notes:
1. Render out any asset from nuke
2. create ExtractReview profiles with slate
3. publish and check timecode in a variety of scenarios
